pkg/connector/datasets.go

```diff
@@ -44,18 +46,32 @@ var (
 		This field will accept any of the above formats, but will return only the legacy format. For example,
 		if you set this field to "roles/bigquery.dataOwner", it will be returned back as "OWNER".
 	*/
-	ownerRole      = string(bigquery.OwnerRole)
-	readerRole     = string(bigquery.ReaderRole)
-	writerRole     = string(bigquery.WriterRole)
-	legacyRolesMap = map[string]string{
-		ownerRole:  "roles/bigquery.dataOwner",
-		readerRole: "roles/bigquery.dataEditor",
-		writerRole: "roles/bigquery.dataViewer",
-	}
+	ownerRole  = string(bigquery.OwnerRole)
+	readerRole = string(bigquery.ReaderRole)
+	writerRole = string(bigquery.WriterRole)
+
 	legacyRolesToEntitlementsMap = map[string]string{
 		ownerRole:  ownerEntitlement,
-		readerRole: writerEntitlement,
-		writerRole: viewerEntitlement,
+		readerRole: viewerEntitlement,
+		writerRole: writerEntitlement,
+	}
+
+	specialGroupNameToPolicyBindingRoleMap = map[string]string{
+		"projectOwners":  "roles/owner",
+		"projectReaders": "roles/viewer",
+		"projectWriters": "roles/editor",
+	}
+
+	iamRoleToEntitlementMap = map[string]string{
+		"roles/bigquery.admin":              writerEntitlement,
+		"roles/bigquery.studioAdmin":        writerEntitlement,
+		"roles/bigquery.user":               viewerEntitlement,
+		"roles/bigquery.resourceEditor":     viewerEntitlement, // Can read and modify dataset metadata but not the dataset content it self.
+		"roles/bigquery.metadataViewer":     viewerEntitlement,
+		"roles/bigquery.filteredDataViewer": viewerEntitlement, // Restricted Read Access. Can only access table rows which match their policy.
+		"roles/admin":                       writerEntitlement,
+		"roles/editor":                      writerEntitlement,
+		"roles/reader":                      viewerEntitlement,
```

Comment:
Should we maybe instead only map OWNER, WRITER and READER, but not map any of these other entitlements, but instead emit them as possible entitlements for the resource?